index_documents.go

```diff
@@ -0,0 +1,162 @@
+package meilisearch
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+func (i Index) GetDocument(identifier string, documentPtr interface{}) error {
+	req := internalRequest{
+		endpoint:            "/indexes/" + i.UID + "/documents/" + identifier,
+		method:              http.MethodGet,
+		withRequest:         nil,
+		withResponse:        documentPtr,
+		acceptedStatusCodes: []int{http.StatusOK},
+		functionName:        "GetDocument",
+	}
+	if err := i.client.executeRequest(req); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (i Index) GetDocuments(request DocumentsRequest, response interface{}) error {
+	req := internalRequest{
+		endpoint:            "/indexes/" + i.UID + "/documents",
+		method:              http.MethodGet,
+		withRequest:         request,
+		withResponse:        response,
+		withQueryParams:     map[string]string{},
+		acceptedStatusCodes: []int{http.StatusOK},
+		functionName:        "GetDocuments",
+	}
+	if request.Limit != 0 {
+		req.withQueryParams["limit"] = strconv.FormatInt(request.Limit, 10)
+	}
```

Comment:
why the condition `request.Limit != 0`?

Reply:
Because in golang there are no optional parameters, all query parameters are passed via structures and checked if they exist. Maybe there is a better way to do it.

Reply:
Ok, that's all right, again de default zero values in Go that force this behavior. 👌🏽